org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2]" [50000-214]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:395)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:125)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:3318)
	at org.h2.mvstore.MVStore.panic(MVStore.java:593)
	at org.h2.mvstore.MVStore.write(MVStore.java:1271)
	at org.h2.mvstore.MVStore.storeBuffer(MVStore.java:1750)
	at org.h2.mvstore.MVStore.lambda$serializeAndStore$5(MVStore.java:1600)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2]" [50000-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:554)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	... 13 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:493)
	at org.h2.mvstore.FileStore.writeFully(FileStore.java:113)
	at org.h2.mvstore.MVStore.write(MVStore.java:1269)
	... 7 more
Caused by: java.io.IOException: No space left on device
	at sun.nio.ch.FileDispatcherImpl.pwrite0(Native Method)
	at sun.nio.ch.FileDispatcherImpl.pwrite(FileDispatcherImpl.java:66)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:89)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.FileChannelImpl.writeInternal(FileChannelImpl.java:772)
	at sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:758)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:489)
	... 9 more
2022-08-31 11:11:58 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: The database has been closed; SQL statement:
insert into fingerprints (path, text, md5) values (?, ?, ?) [90098-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:678)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:159)
	at org.h2.mvstore.db.MVTable.convertException(MVTable.java:687)
	at org.h2.mvstore.db.MVSecondaryIndex.add(MVSecondaryIndex.java:190)
	at org.h2.mvstore.db.MVTable.addRow(MVTable.java:519)
	at org.h2.command.dml.Insert.insertRows(Insert.java:174)
	at org.h2.command.dml.Insert.update(Insert.java:135)
	at org.h2.command.dml.DataChangeStatement.update(DataChangeStatement.java:74)
	at org.h2.command.CommandContainer.update(CommandContainer.java:169)
	at org.h2.command.Command.executeUpdate(Command.java:252)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:209)
	at org.h2.jdbc.JdbcPreparedStatement.executeBatchElement(JdbcPreparedStatement.java:1317)
	at org.h2.jdbc.JdbcPreparedStatement.executeBatch(JdbcPreparedStatement.java:1263)
	at scala.meta.internal.metals.Fingerprints.save(Fingerprints.scala:45)
	at scala.meta.internal.metals.MetalsLanguageServer.shutdown(MetalsLanguageServer.scala:1029)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	at org.eclipse.lsp4j.jsonrpc.serviceorg.h2.jdbc.JdbcSQLNonTransientConnectionException: The database has been closed [90098-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:678)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.engine.SessionLocal.getTransaction(SessionLocal.java:1596)
	at org.h2.engine.SessionLocal.getStatementSavepoint(SessionLocal.java:1606)
	at org.h2.engine.SessionLocal.setSavepoint(SessionLocal.java:848)
	at org.h2.command.Command.executeUpdate(Command.java:244)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:209)
	at org.h2.jdbc.JdbcPreparedStatement.executeBatchElement(JdbcPreparedStatement.java:1317)
	at org.h2.jdbc.JdbcPreparedStatement.executeBatch(JdbcPreparedStatement.java:1263)
	at scala.meta.internal.metals.Fingerprints.save(Fingerprints.scala:45)
	at scala.meta.internal.metals.MetalsLanguageServer.shutdown(MetalsLanguageServer.scala:1029)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: org.h2.mvstore.MVStoreException: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2] [2.1.214/3] [2.1.214/3]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:1529)
	at org.h2.mvstore.MVStore.store(MVStore.java:1496)
	at org.h2.mvstore.MVStore.commit(MVStore.java:1469)
	at org.h2.mvstore.MVStore.beforeWrite(MVStore.java:2890)
	at org.h2.mvstore.MVMap.beforeWrite(MVMap.java:969)
	at org.h2.mvstore.MVMap.operate(MVMap.java:1757)
	at org.h2.mvstore.MVMap.rewritePage(MVMap.java:691)
	at org.h2.mvstore.MVStore.rewriteChunks(MVStore.java:2523)
	at org.h2.mvstore.MVStore.compactRewrite(MVStore.java:2501)
	at org.h2.mvstore.MVStore.rewriteChunks(MVStore.java:2340)
	at org.h2.mvstore.MVStore.writeInBackground(MVStore.java:3242)
	at org.h2.mvstore.MVStore$BackgroundWriterThread.run(MVStore.java:3775)
Caused by: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2] [2.1.214/3]
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at org.h2.mvstore.MVStore.submitOrRun(MVStore.java:1541)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:1517)
	... 11 more
Caused by: org.h2.mvstore.MVStoreException: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2] [2.1.214/3]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2]
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at org.h2.mvstore.MVStore.submitOrRun(MVStore.java:1541)
	at org.h2.mvstore.MVStore.serializeAndStore(MVStore.java:1600)
	... 6 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:493)
	at org.h2.mvstore.FileStore.writeFully(FileStore.java:113)
	at org.h2.mvstore.MVStore.write(MVStore.java:1269)
	at org.h2.mvstore.MVStore.storeBuffer(MVStore.java:1750)
	at org.h2.mvstore.MVStore.lambda$serializeAndStore$5(MVStore.java:1600)
	... 5 more
Caused by: java.io.IOException: No space left on device
	at sun.nio.ch.FileDispatcherImpl.pwrite0(Native Method)
	at sun.nio.ch.FileDispatcherImpl.pwrite(FileDispatcherImpl.java:66)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:89)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.FileChannelImpl.writeInternal(FileChannelImpl.java:772)
	at sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:758)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:489)
	... 9 more
2022-08-31 11:11:58 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: The database has been closed [90098-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:678)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.engine.SessionLocal.getTransaction(SessionLocal.java:1596)
	at org.h2.engine.SessionLocal.getStatementSavepoint(SessionLocal.java:1606)
	at org.h2.engine.SessionLocal.setSavepoint(SessionLocal.java:848)
	at org.h2.command.Command.executeUpdate(Command.java:244)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:209)
	at org.h2.jdbc.JdbcPreparedStatement.executeBatchElement(JdbcPreparedStatement.java:1317)
	at org.h2.jdbc.JdbcPreparedStatement.executeBatch(JdbcPreparedStatement.java:1263)
	at scala.meta.internal.metals.Fingerprints.save(Fingerprints.scala:45)
	at scala.meta.internal.metals.MetalsLanguageServer.shutdown(MetalsLanguageServer.scala:1029)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.T2022-08-31 11:11:58 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: The database has been closed [90098-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:678)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.engine.SessionLocal.getTransaction(SessionLocal.java:1596)
	at org.h2.engine.SessionLocal.getStatementSavepoint(SessionLocal.java:1606)
	at org.h2.engine.SessionLocal.setSavepoint(SessionLocal.java:848)
	at org.h2.command.Command.executeUpdate(Command.java:244)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:209)
	at org.h2.jdbc.JdbcPreparedStatement.executeBatchElement(JdbcPreparedStatement.java:1317)
	at org.h2.jdbc.JdbcPreparedStatement.executeBatch(JdbcPreparedStatement.java:1263)
	at scala.meta.internal.metals.Fingerprints.save(Fingerprints.scala:45)
	at scala.meta.internal.metals.MetalsLanguageServer.shutdown(MetalsLanguageServer.scala:1029)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: org.h2.mvstore.MVStoreException: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2] [2.1.214/3] [2.1.214/3]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:1529)
	at org.h2.mvstore.MVStore.store(MVStore.java:1496)
	at org.h2.mvstore.MVStore.commit(MVStore.java:1469)
	at org.h2.mvstore.MVStore.beforeWrite(MVStore.java:2890)
	at org.h2.mvstore.MVMap.beforeWrite(MVMap.java:969)
	at org.h2.mvstore.MVMap.operate(MVMap.java:1757)
	at org.h2.mvstore.MVMap.rewritePage(MVMap.java:691)
	at org.h2.mvstore.MVStore.rewriteChunks(MVStore.java:2523)
	at org.h2.mvstore.MVStore.compactRewrite(MVStore.java:2501)
	at org.h2.mvstore.MVStore.rewriteChunks(MVStore.java:2340)
	at org.h2.mvstore.MVStore.writeInBackground(MVStore.java:3242)
	at org.h2.mvstore.MVStore$BackgroundWriterThread.run(MVStore.java:3775)
Caused by: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2] [2.1.214/3]
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at org.h2.mvstore.MVStore.submitOrRun(MVStore.java:1541)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:1517)
	... 11 more
Caused by: org.h2.mvstore.MVStoreException: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2] [2.1.214/3]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.MVStore.serializeAndStore(MVStore.java:1605)
	at org.h2.mvstore.MVStore.lambda$storeNow$4(MVStore.java:1518)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2]
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at org.h2.mvstore.MVStore.submitOrRun(MVStore.java:1541)
	at org.h2.mvstore.MVStore.serializeAndStore(MVStore.java:1600)
	... 6 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:493)
	at org.h2.mvstore.FileStore.writeFully(FileStore.java:113)
	at org.h2.mvstore.MVStore.write(MVStore.java:1269)
	at org.h2.mvstore.MVStore.storeBuffer(MVStore.java:1750)
	at org.h2.mvstore.MVStore.lambda$serializeAndStore$5(MVStore.java:1600)
	... 5 more
Caused by: java.io.IOException: No space left on device
	at sun.nio.ch.FileDispatcherImpl.pwrite0(Native Method)
	at sun.nio.ch.FileDispatcherImpl.pwrite(FileDispatcherImpl.java:66)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:89)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.FileChannelImpl.writeInternal(FileChannelImpl.java:772)
	at sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:758)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:489)
	... 9 more
2022-08-31 11:11:58 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: The database has been closed [90098-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:678)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.engine.SessionLocal.getTransaction(SessionLocal.java:1596)
	at org.h2.engine.SessionLocal.getStatementSavepoint(SessionLocal.java:1606)
	at org.h2.engine.SessionLocal.setSavepoint(SessionLocal.java:848)
	at org.h2.command.Command.executeUpdate(Command.java:244)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:209)
	at org.h2.jdbc.JdbcPreparedStatement.executeBatchElement(JdbcPreparedStatement.java:1317)
	at org.h2.jdbc.JdbcPreparedStatement.executeBatch(JdbcPreparedStatement.java:1263)
	at scala.meta.internal.metals.Fingerprints.save(Fingerprints.scala:45)
	at scala.meta.internal.metals.MetalsLanguageServer.shutdown(MetalsLanguageServer.scala:1029)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: org.h2.mvstore.MVStoreException: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2] [2.1.214/3] [2.1.214/3]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:1529)
	at org.h2.mvstore.MVStore.store(MVStore.java:1496)
	at org.h2.mvstore.MVStore.commit(MVStore.java:1469)
	at org.h2.mvstore.MVStore.beforeWrite(MVStore.java:2890)
	at org.h2.mvstore.MVMap.beforeWrite(MVMap.java:969)
	at org.h2.mvstore.MVMap.operate(MVMap.java:1757)
	at org.h2.mvstore.MVMap.rewritePage(MVMap.java:691)
	at org.h2.mvstore.MVStore.rewriteChunks(MVStore.java:2523)
	at org.h2.mvstore.MVStore.compactRewrite(MVStore.java:2501)
	at org.h2.mvstore.MVStore.rewriteChunks(MVStore.java:2340)
	at org.h2.mvstore.MVStore.writeInBackground(MVStore.java:3242)
	at org.h2.mvstore.MVStore$BackgroundWriterThread.run(MVStore.java:3775)
Caused by: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2] [2.1.214/3]
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at org.h2.mvstore.MVStore.submitOrRun(MVStore.java:1541)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:1517)
	... 11 more
Caused by: org.h2.mvstore.MVStoreException: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2] [2.1.214/3]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.MVStore.serializeAndStore(MVStore.java:1605)
	at org.h2.mvstore.MVStore.lambda$storeNow$4(MVStore.java:1518)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.util.concurrent.ExecutionException: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2]
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at org.h2.mvstore.MVStore.submitOrRun(MVStore.java:1541)
	at org.h2.mvstore.MVStore.serializeAndStore(MVStore.java:1600)
	... 6 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@2614d2a failed; length 12267520 at 64471040 [2.1.214/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:493)
	at org.h2.mvstore.FileStore.writeFully(FileStore.java:113)
	at org.h2.mvstore.MVStore.write(MVStore.java:1269)
	at org.h2.mvstore.MVStore.storeBuffer(MVStore.java:1750)
	at org.h2.mvstore.MVStore.lambda$serializeAndStore$5(MVStore.java:1600)
	... 5 more
Caused by: java.io.IOException: No space left on device
	at sun.nio.ch.FileDispatcherImpl.pwrite0(Native Method)
	at sun.nio.ch.FileDispatcherImpl.pwrite(FileDispatcherImpl.java:66)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:89)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.FileChannelImpl.writeInternal(FileChannelImpl.java:772)
	at sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:758)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:489)
	... 9 more
2022-08-31 11:11:58 jd2022-09-12 13:58:42 database: flush
org.h2.message.DbException: OutOfMemoryError
	at org.h2.message.DbException.<clinit>(DbException.java:64)
	at org.h2.message.TraceObject.<clinit>(TraceObject.java:108)
	at org.h2.Driver.connect(Driver.java:59)
	at org.flywaydb.core.internal.jdbc.DriverDataSource.getConnectionFromDriver(DriverDataSource.java:258)
	at org.flywaydb.core.internal.jdbc.DriverDataSource.getConnection(DriverDataSource.java:223)
	at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:48)
	at org.flywaydb.core.internal.jdbc.JdbcConnectionFactory.<init>(JdbcConnectionFactory.java:75)
	at org.flywaydb.core.FlywayExecutor.execute(FlywayExecutor.java:146)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:126)
	at scala.meta.internal.metals.Tables.migrateOrRestart(Tables.scala:152)
	at scala.meta.internal.metals.Tables.tryUrl(Tables.scala:144)
	at scala.meta.internal.metals.Tables.persistentConnection(Tables.scala:138)
	at scala.meta.internal.metals.Tables.tryAutoServer(Tables.scala:81)
	at scala.meta.internal.metals.Tables.connect(Tables.scala:48)
	at scala.meta.internal.metals.Tables.connection(Tables.scala:74)
	at scala.meta.internal.metals.Tables.$anonfun$fingerprints$1(Tables.scala:39)
	at scala.meta.internal.metals.Fingerprints.load(Fingerprints.scala:52)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$initialize$2(MetalsLanguageServer.scala:856)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:678)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:467)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.sql.SQLException: OutOfMemoryError
	at org.h2.message.DbException.<clinit>(DbException.java:62)
	... 22 more
Caused by: java.lang.OutOfMemoryError
	... 23 more
2022-09-12 13:58:49 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Out of memory. [90108-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:678)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.engine.Database.throwLastBackgroundException(Database.java:1782)
	at org.h2.engine.SessionLocal.beforeCommitOrRollback(SessionLocal.java:754)
	at org.h2.engine.SessionLocal.commit(SessionLocal.java:668)
	at org.h2.command.Command.stop(Command.java:157)
	at org.h2.command.CommandContainer.stop(CommandContainer.java:259)
	at org.h2.command.Command.executeQuery(Command.java:229)
	at org.h2.jdbc.JdbcPreparedStatement.executeQuery(JdbcPreparedStatement.java:128)
	at scala.meta.internal.metals.JdbcEnrichments$XtensionConnection.query(JdbcEnrichments.scala:37)
	at scala.meta.internal.metals.DependencySources.getBuildTarget(DependencySources.scala:27)
	at scala.meta.internal.metals.BuildTargets.$anonfun$inverseSources$2(BuildTargets.scala:216)
	at scala.Option.flatMap(Option.scala:283)
	at scala.meta.internal.metals.BuildTargets.inverseSources(BuildTargets.scala:216)
	at scala.meta.internal.metals.FileSystemSemanticdbs.textDocument(FileSystemSemanticdbs.scala:33)
	at scala.meta.internal.metals.AggregateSemanticdbs.loop$1(AggregateSemanticdbs.scala:30)
	at scala.meta.internal.metals.AggregateSemanticdbs.textDocument(AggregateSemanticdbs.scala:36)
	at scala.meta.internal.metals.DocumentHighlightProvider.documentHighlight(DocumentHighlightProvider.scala:27)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$documentHighlights$1(MetalsLanguageServer.scala:1508)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$apply$2(CancelTokens.scala:26)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:678)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:467)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: org.h2.message.DbException: OutOfMemoryError
	at org.h2.message.DbException.<clinit>(DbException.java:64)
	at org.h2.message.TraceObject.<clinit>(TraceObject.java:108)
	at org.h2.Driver.connect(Driver.java:59)
	at org.flywaydb.core.internal.jdbc.DriverDataSource.getConnectionFromDriver(DriverDataSource.java:258)
	at org.flywaydb.core.internal.jdbc.DriverDataSource.getConnection(DriverDataSource.java:223)
	at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:48)
	at org.flywaydb.core.internal.jdbc.JdbcConnectionFactory.<init>(JdbcConnectionFactory.java:75)
	at org.flywaydb.core.FlywayExecutor.execute(FlywayExecutor.java:146)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:126)
	at scala.meta.internal.metals.Tables.migrateOrRestart(Tables.scala:152)
	at scala.meta.internal.metals.Tables.tryUrl(Tables.scala:144)
	at scala.meta.internal.metals.Tables.persistentConnection(Tables.scala:138)
	at scala.meta.internal.metals.Tables.tryAutoServer(Tables.scala:81)
	at scala.meta.internal.metals.Tables.connect(Tables.scala:48)
	at scala.meta.internal.metals.Tables.connection(Tables.scala:74)
	at scala.meta.internal.metals.Tables.$anonfun$fingerprints$1(Tables.scala:39)
	at scala.meta.internal.metals.Fingerprints.load(Fingerprints.scala:52)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$initialize$2(MetalsLanguageServer.scala:856)
	... 5 more
Caused by: java.sql.SQLException: OutOfMemoryError
	at org.h2.message.DbException.<clinit>(DbException.java:62)
	... 22 more
Caused by: java.lang.OutOfMemoryError
	... 23 more
